src/ScriptBuilder/Saga/SagaDefinitionValidator.cs

```diff
@@ -14,9 +14,24 @@ internal static void ValidateSagaDefinition(string correlationProperty, string s
             throw new ErrorsException($"The Saga '{sagaName}' has an empty string defined for TransitionalCorrelationProperty.");
         }
 
-        if (tableSuffix != null && string.IsNullOrWhiteSpace(tableSuffix))
+        if (tableSuffix != null)
         {
-            throw new ErrorsException($"The Saga '{sagaName}' has an empty string defined for to TableSuffix.");
+            if (string.IsNullOrWhiteSpace(tableSuffix))
+            {
+                throw new ErrorsException($"The Saga '{sagaName}' has an empty string defined for to TableSuffix.");
+            }
+            if (tableSuffix.Contains("]"))
+            {
+                throw new ErrorsException($"The Saga '{sagaName}' has a tableSuffix that contains a ']'. Delimiters are automatically added and are not required in configuration.");
+            }
+            if (tableSuffix.Contains("["))
+            {
+                throw new ErrorsException($"The Saga '{sagaName}' has a tableSuffix that contains a '['. Delimiters are automatically and are not required in configuration.");
+            }
+            if (tableSuffix.Contains("`"))
+            {
+                throw new ErrorsException($"The Saga '{sagaName}' has a tableSuffix that contains a '`'. Delimiters are automatically and are not required in configuration.");
```

Comment:
Also, I think it is better to say that the names are *automatically quoted* instead of saying we add delimiter. I would be confused as I would tread the `.` as a delimiter in the fully-qualified name, not the `[]`s